LibVLCSharp/Shared/Media.cs

```diff
@@ -526,6 +526,19 @@ MediaEventManager EventManager
         /// </summary>
         public void Parse() => Native.LibVLCMediaParse(NativeReference);
 
+        /// <summary>
+        /// Parse the media asynchronously with options. 
+        /// To track when this is over you can listen to the ParsedChanged event. However if this functions returns an error, you will not receive any events.
+        /// It uses a flag to specify parse options (see <see cref="MediaParseOptions"/>). All these flags can be combined. By default, media is parsed if it's a local file.
+        /// <para/> Note: Parsing can be aborted with ParseStop().
+        /// </summary>
+        /// <param name="options">parse options</param>
+        /// <param name="timeout">maximum time allowed to preparse the media. 
+        /// <para/>If -1, the default "preparse-timeout" option will be used as a timeout. 
+        /// <para/>If 0, it will wait indefinitely. If > 0, the timeout will be used (in milliseconds). 
+        /// </param>
+        public void Parse(MediaParseOptions options, int timeout = -1) => Native.LibVLCMediaParseWithOptions(NativeReference, options, timeout);
```

Comment:
Can we make it a Task? that would be easier to call in C#

Reply:
There is a plan to add additional `ParseAsync` APIs that would return a `Task`. However:
- We should expose both, and the sync API is the easiest to expose right now. Would be nice to ship this so I can unblock https://code.videolan.org/videolan/LibVLCSharp/issues/108.
- Exposing a Task means we have to know for sure when to set the Task status to Completed. But the doc states
```
To track when this is over you can listen to libvlc_MediaParsedChanged event. However if this functions returns an error, you will not receive any events.
```
Which makes me rather uncertain about the async implementation of this. Hence why I'd like to merge this first and figure out the async stuff in an another PR.

Reply:
Unless that just means to check the return value in which case, that should be easy...?
Regardless, I'd rather do it in a separate PR.

Reply:
this method is async by nature. As I understand the comment, either it fails immediately and returns an error code, or you will be assured to have a corresponding parsing event